core_impl/src/main/java/com/google/instrumentation/trace/TraceComponentImplBase.java

```diff
@@ -14,11 +14,14 @@
 package com.google.instrumentation.trace;
 
 /** Implementation of the {@link TraceComponent}. */
-public final class TraceComponentImpl extends TraceComponent {
+public class TraceComponentImplBase extends TraceComponent {
```

Comment:
StatsManagerImplBase isn't abstract, because it is used for testing in core_impl.  For example, [StatsManagerImplTest](https://github.com/google/instrumentation-java/blob/0c9fcffe501ecf6e583d917159f41c72c81d8841/core_impl/src/test/java/com/google/instrumentation/stats/StatsManagerImplTest.java#L58) creates a StatsManagerImplBase with a test clock and event queue and then tests the functionality that doesn't depend on Java version.  I assumed that the core_impl tests would also need to create TraceComponentImpls.